build-coordinator/src/main/java/org/jboss/pnc/core/builder/BuildScheduler.java

```diff
@@ -29,5 +29,6 @@
  * @author <a href="mailto:matejonnet@gmail.com">Matej Lazar</a>
  */
 public interface BuildScheduler {
+    boolean hasBeenBuilt(BuildTask buildTask);
```

Comment:
I'm not entirely sure if this is the right approach... consider the following use cases:

1. Each scheduler might use different policies to check whether a BC has been built. For example I can imagine, that BPM might check additionally Brew (or any other repository) whereas Local builder might check only DB.

2. BC should be checked just *before* the build (think about building 2 products with the same BC built concurrently). This can only be done at scheduler level. BuildCoordinator just takes the BuildTask and passes it to the scheduler.

To wrap it up - if I move this check to the build coordinator, we might not be able to meet those 2 use cases. The question is - is it ok?